v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dao/MySqlDao.java

```diff
@@ -90,9 +90,53 @@ public void batchWrite(List<String> batchStatements) throws SQLException {
         status = true;
       } catch (com.mysql.cj.jdbc.exceptions.CommunicationsException e) {
         // TODO: retry handling is configurable with retry count
-        LOG.warn("Connection exception while executing SQL, will retry : " + e.getMessage());
+        LOG.warn(
+            "Connection exception while executing SQL for shard : "
+                + shardId
+                + ", will retry : "
+                + e.getMessage());
         // gives indication that the shard is being retried
         Metrics.counter(MySqlDao.class, "mySQL_retry_" + shardId).inc();
+        try {
+          Thread.sleep(1000);
+        } catch (java.lang.InterruptedException ex) {
+          throw new RuntimeException(ex);
+        }
```

Comment:
I didn't get what we are trying to do -

If we get `CommunicationsException` , we will - 

1. Log a WARN
2. Increment a counter
3. Try to sleep for 1s

Why are we trying to sleep for 1s here? 
There should be a `backoff/retryable` `client` implemented here right? 

One which performs retries and backoffs for the `batchWrite` based on a configuration?

Reply:
This is by design

Reply:
I have taken some default here, don't want to over-engineer and take configuration for such things at this stage. If needed will add in future.

Reply:
Understood, why does the `Thread.Sleep` happen inside the `batchWrite` method though? Isn't this similar having a client which contains methods which map to external APIs?

The client would configure the behaviour and method corresponding to the API would do the mark. 

Here it seems like the method which maps to the API is aware of the retry logic?

Please correct me if I am wrong!

Reply:
Did not understand the comment above. I want the caller of batchWrite method to not worry about the retry. There are specific exceptions visible to batchWrite and it can decide to retry based on those.